libgadget/exchange.c

```diff
@@ -52,7 +67,7 @@ _transpose_plan_entries(ExchangePlanEntry * entries, int * count, int ptype)
         }
     }
 }
-int domain_exchange(int (*layoutfunc)(int p)) {
+int domain_exchange(int (*layoutfunc)(int p), int do_gc) {
```

Comment:
We prefer running gc inside exchange because it reduces the maximum memory footprint.
`int do_gc` really is a `incremental_hint` -- a flag to notify the routine not to worry about reducing the peak memory footprint.

Reply:
Right. At the moment do_gc is true unless we know for sure that a gc will be run just after the exchange by the full domain decomposition. So all it does right now is avoid doing a gc twice.

But we can later use it as a hint to skip reducing the the peak memory when exchange is small.

Reply:
Yes. But it is surprising when `do_gc` is True, gc is not always done.

Reply:
I think you are misreading: shall_we_gc is an or, not an and. When do_gc is True, gc is always done. When do_gc is false, gc may or may not be done.

Reply:
Ah. Indeed! I misread.